seqr/views/apis/staff_api_tests.py

```diff
@@ -531,3 +570,39 @@ def test_anvil_export(self, mock_export_multiple_files):
         self.assertIn(EXPECTED_PI_SAMPLE_ROW, exported_files[1][2])
         self.assertIn(EXPECTED_PI_FAMILY_ROW, exported_files[2][2])
         self.assertIn(EXPECTED_PI_DISCOVERY_ROW, exported_files[3][2])
+
+    @responses.activate
+    def test_sample_metadata_export(self):
+        url = reverse(sample_metadata_export, args=[PROJECT_GUID])
+        _check_login(self, url)
+
+        responses.add(responses.GET, '{}/Samples'.format(AIRTABLE_URL),
+                      json=AIRTABLE_SAMPLE_RECORDS, status=200)
+        responses.add(responses.GET, '{}/Collaborator'.format(AIRTABLE_URL),
+                      json=AIRTABLE_COLLABORATOR_RECORDS, status=200)
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+        response_json = response.json()
+        self.assertListEqual(response_json.keys(), ['rows'])
+        self.assertIn(EXPECTED_SAMPLE_METADATA_ROW, response_json['rows'])
+
+    def test_saved_variants_page(self):
+        url = reverse(saved_variants_page, args=['Tier 1 - Novel gene and phenotype'])
+        _check_login(self, url)
+
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+        response_json = response.json()
+        self.assertListEqual(response_json.keys(), ['projectsByGuid', 'locusListsByGuid', 'savedVariantsByGuid', 'variantFunctionalDataByGuid', 'genesById', 'variantNotesByGuid', 'individualsByGuid', 'variantTagsByGuid', 'familiesByGuid'])
+
+    @mock.patch('seqr.views.apis.staff_api.file_iter')
+    def test_upload_qc_pipeline_output(self, mock_file_iter):
+        url = reverse(upload_qc_pipeline_output,)
+        _check_login(self, url)
+
+        mock_file_iter.return_value = SAMPLE_QC_DATA
+        response = self.client.post(url, content_type='application/json',
+                data=json.dumps({'file': 'gs://seqr-datasets/v02/GRCh38/RDG_WES_Broad_Internal/v15/sample_qc/final_output/seqr_sample_qc.tsv'}))
+        self.assertEqual(response.status_code, 200)
+        response_json = response.json()
+        self.assertListEqual(response_json.keys(), ['info', 'errors', 'warnings'])
```

Comment:
You need to test the content of all of those keys, as well as test that the expected Individual models got updated

Reply:
Okay